Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -25,6 +25,15 @@
 
 namespace RDKit {
 
+#ifdef RDK_THREADSAFE_SSS
+namespace {
+std::mutex & smiles_setprop_locker() {
```

Comment:
We could just have
```
namespace {
  std::mutex ...
}
```

but this prevents the mutex creation until it is needed.  However, I expect it is always needed in practice.